### ISSUE TITLE
How can I create a custom Task ID and push to redis

### ISSUE THREAD

**Issue by DilanJT** [geofmureithi/apalis#415]
### Discussed in https://github.com/geofmureithi/apalis/discussions/414

<div type='discussions-op-text'>

<sup>Originally posted by **DilanJT** September  9, 2024</sup>

```
let task_id = match storage.push(updated_job.clone()).await {
        Ok(task_id) => task_id,
        Err(e) => {
            println!("Error pushing job to storage: {}", e);
            return Err(Box::new(e));
        }
    };
    
```
    Instead of getting a generated task_id, I want to give a custom task_id along with storage.push()
    
</div>

**Comment by AzHicham** [comment]
+1 This could be a great feature for integration tests, to be used with fetch_by_id(job_id: &TaskId)

**Comment by geofmureithi** [comment]
This will be resolved by #416 as it introduces methods like `push_request` and `schedule_request`.
See an example here https://github.com/geofmureithi/apalis/pull/416/files#diff-d019ef5f40fdc1d870b236e2170b65460839706241f35d6716359f085f9738bdR30

**Comment by geofmureithi** [comment]
This is resolved in the latest release candidate.
Basically, you can use `push_request` instead of `push`
